packages/plugin-ext/src/hosted/node/hosted-plugins-manager.ts

```diff
@@ -0,0 +1,149 @@
+     * Chacks if watcher script to recomple plugin is running.
+     *
+     * @param uri uri to plugin root folder.
+     */
+    isWatchCompilationRunning(uri: string): Promise<boolean>;
+}
+
+@injectable()
+export class HostedPluginsManagerImpl implements HostedPluginsManager {
+
+    @inject(HostedPluginSupport)
+    protected readonly hostedPluginSupport: HostedPluginSupport;
+
+    protected watchCompilationRegistry: Map<string, cp.ChildProcess>;
+
+    constructor() {
+        this.watchCompilationRegistry = new Map();
+    }
+
+    runWatchCompilation(uri: string): Promise<void> {
+        const pluginRootPath = this.getFsPath(uri);
+
+        if (this.watchCompilationRegistry.has(pluginRootPath)) {
+            throw new Error('Watcher is already running in ' + pluginRootPath);
+        }
+
+        if (!this.checkWatchScript(pluginRootPath)) {
+            this.hostedPluginSupport.sendLog({
+                data: 'Plugin in ' + uri + ' doesn\'t have watch script',
+                type: LogType.Error
+            });
+            throw new Error('Watch script doesn\'t exist in ' + pluginRootPath + 'package.json');
+        }
+
+        return this.runWatchScript(pluginRootPath);
+    }
+
+    stopWatchCompilation(uri: string): Promise<void> {
+        const pluginPath = this.getFsPath(uri);
+
+        const watchProcess = this.watchCompilationRegistry.get(pluginPath);
+        if (!watchProcess) {
+            throw new Error('Watcher is not running in ' + pluginPath);
+        }
+
+        watchProcess.kill();
+        return Promise.resolve();
+    }
+
+    isWatchCompilationRunning(uri: string): Promise<boolean> {
+        const pluginPath = this.getFsPath(uri);
+
+        return new Promise(resolve => resolve(this.watchCompilationRegistry.has(pluginPath)));
+    }
+
+    protected runWatchScript(path: string): Promise<void> {
+        const watchProcess = cp.spawn('yarn', ['run', 'watch'], { cwd: path });
+        watchProcess.on('exit', () => this.unregisterWatchScript(path));
+
+        this.watchCompilationRegistry.set(path, watchProcess);
+        this.hostedPluginSupport.sendLog({
+            data: 'Compilation watcher has been started in ' + path,
+            type: LogType.Info
+        });
+        return Promise.resolve();
+    }
+
+    protected unregisterWatchScript(path: string) {
+        this.watchCompilationRegistry.delete(path);
+        this.hostedPluginSupport.sendLog({
+            data: 'Compilation watcher has been stopped in ' + path,
+            type: LogType.Info
+        });
+    }
+
+    /**
+     * Checks whether watch script is present into package.json by given parent folder.
+     *
+     * @param pluginPath path to plugin's root directory
+     */
+    protected checkWatchScript(pluginPath: string): boolean {
+        const pluginPackageJsonPath = pluginPath + 'package.json';
+        if (fs.existsSync(pluginPackageJsonPath)) {
+            const packageJson = require(pluginPackageJsonPath);
```

Comment:
we don't have the model already parsed somewhere ?

Reply:
@benoitf no, I cannot find it. As about including.... I have doubts, because it is dynamic and could be changed during development process. WDYT?